src/platform/pantherlake/include/platform/drivers/interrupt.h

```diff
@@ -0,0 +1,32 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022-2024 Intel Corporation.
+ *
+ * Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
+ *         Keyon Jie <yang.jie@linux.intel.com>
+ *         Rander Wang <rander.wang@intel.com>
+ */
+
+#ifndef __SOF_DRIVERS_INTERRUPT_H__
+#error "This file shouldn't be included from outside of sof/drivers/interrupt.h"
+#endif
+
+#ifndef __PLATFORM_DRIVERS_INTERRUPT_H__
+#define __PLATFORM_DRIVERS_INTERRUPT_H__
+
+#if !defined(__ASSEMBLER__) && !defined(LINKER)
+
+#include <ace/drivers/interrupt.h>
+
+#include <rtos/bit.h>
+
+#endif
+
+/* Required by sof/drivers/interrupt.h */
+#define PLATFORM_IRQ_CHILDREN	32
+
+/* Required by zephyr/wrapper.c */
+#define IRQ_NUM_EXT_LEVEL2	4	/* level 2 */
+#define IRQ_NUM_EXT_LEVEL5	16	/* level 5 */
```

Comment:
100% confused by this file.  Zephyr's handling the interrupts and I can seem them declared in DTS in the ace30 submission.  I can understand if the file is a legacy thing that needs to be included, or if it's the only source of some preprocessor macro that someone else uses.  But why are we putting interrupt interrupt routing information here, when it really shouldn't be used anywhere?

Reply:
@andyross See my comment on this PR, this file (and all platfrom/drivers/*.h) can be dropped completely. This longtail of SOF-side support for building with both XTOS and Zephyr (see https://github.com/thesofproject/sof/issues/5794) and mostly cleaned up. In SOF2.10 cycle (Q2 this year), we've cleaned up these further, but there is still some work left in the platform layer (as we still have HW in SOF that does not support Zephyr -- as long as we have some, we need to carry some abstractions with us)